src/VAC/View.cpp

```diff
@@ -1087,46 +1093,221 @@ void View::PMRReleaseEvent(int action, double x, double y)
+        double centerY = (topY + bottomY) / 2;
+
+        double startX = centerX + radiusW;
+        double startY = centerY;
+        vac_->beginSketchEdge(startX, startY, w, interactiveTime());
+
+        for (double deg = 0; deg < M_PI * 2; deg += 0.01)
+        {
+            double x = radiusW * cos(deg);
+            double y = radiusH * sin(deg);
+
+            double newX = centerX + x;
+            double newY = centerY + y;
+            vac_->continueSketchEdge(newX, newY, w);
+        }
+        vac_->endSketchEdge();
+
+        emit allViewsNeedToUpdatePicking();
+        updateHoveredObject(mouse_Event_X_, mouse_Event_Y_);
+        emit allViewsNeedToUpdate();
+    }
+    else if(action == TRIANGLE_ACTION)
+    {
+        vac_->endRectangleOfSelection();
+        lastMousePos_ = QPoint(mouse_Event_X_,mouse_Event_Y_);
+
+        vac_->beginSketchEdge(leftX, bottomY, w, interactiveTime());
+        vac_->continueSketchEdge(leftX + shapeWidth / 2, topY, w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(leftX + shapeWidth / 2, topY, w, interactiveTime());
+        vac_->continueSketchEdge(rightX, bottomY, w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(rightX, bottomY, w, interactiveTime());
+        vac_->continueSketchEdge(leftX, bottomY, w);
+        vac_->endSketchEdge();
+
+        emit allViewsNeedToUpdatePicking();
+        updateHoveredObject(mouse_Event_X_, mouse_Event_Y_);
+        emit allViewsNeedToUpdate();
+    }
+    else if(action == RHOMBUS_ACTION)
+    {
+        vac_->endRectangleOfSelection();
+        lastMousePos_ = QPoint(mouse_Event_X_,mouse_Event_Y_);
+
+        vac_->beginSketchEdge(leftX, topY + shapeHeight / 2, w, interactiveTime());
+        vac_->continueSketchEdge(leftX + shapeWidth / 2, topY, w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(leftX + shapeWidth / 2, topY, w, interactiveTime());
+        vac_->continueSketchEdge(rightX, topY + shapeHeight / 2, w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(rightX, topY + shapeHeight / 2, w, interactiveTime());
+        vac_->continueSketchEdge(leftX + shapeWidth / 2, bottomY, w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(leftX + shapeWidth / 2, bottomY, w, interactiveTime());
+        vac_->continueSketchEdge(leftX, topY + shapeHeight / 2, w);
+        vac_->endSketchEdge();
+
+        emit allViewsNeedToUpdatePicking();
+        updateHoveredObject(mouse_Event_X_, mouse_Event_Y_);
+        emit allViewsNeedToUpdate();
+    }
+    else if(action == PENTAGON_ACTION)
+    {
+        vac_->endRectangleOfSelection();
+        lastMousePos_ = QPoint(mouse_Event_X_,mouse_Event_Y_);
+
+        vac_->beginSketchEdge(getX(36), getY(36), w, interactiveTime());
+        vac_->continueSketchEdge(getX(36 + 72), getY(36 + 72), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX(36 + 72), getY(36 + 72), w, interactiveTime());
```

Comment:
I think it would be nicer to make 36 and 72 into constexpr variables with descriptive names for clarifying this code since they show up so many times. Unclear to me what they actually mean.

Reply:
You are right.
This numbers are values of angle in degree for calculate the coordinates of the vertex of the each polygons. But in future I plan to improve the drawing algorithms(using loops) and it makes sence to done this after changes algorithms.

Reply:
Sure. Best to create a Jira task for it so it's easier to track and not forgotten.

Reply:
Don't worry. I wil not forget about this and I think that Jira task is not needed for this :). But If you want then please create the Jira ticket...